plugins/LilyPad/LilyPad.cpp

```diff
@@ -1209,256 +1212,291 @@ u8 CALLBACK PADpoll(u8 value)
     if (query.lastByte == 0) {
         query.lastByte++;
         query.currentCommand = value;
-        switch (value) {
-            // CONFIG_MODE
-            case 0x43:
-                if (pad->config) {
-                    if (pad->mode == MODE_DIGITAL && config.padConfigs[query.port][query.slot].autoAnalog && !ps2e) {
-                        pad->mode = MODE_ANALOG;
-                    }
-                    // In config mode.  Might not actually be leaving it.
-                    SET_RESULT(ConfigExit);
-                    DEBUG_OUT(0xF3);
-                    return 0xF3;
-                }
-            // READ_DATA_AND_VIBRATE
-            case 0x42:
-                query.response[2] = 0x5A;
-                {
-                    Update(query.port, query.slot);
-                    ButtonSum *sum = &pad->sum;
-
-                    if (config.padConfigs[query.port][query.slot].type == MousePad) {
-                        u8 b1 = 0xFC;
-                        if (sum->buttons[9] > 0) // Left button
-                            b1 -= 8;
-                        if (sum->buttons[10] > 0) // Right button
-                            b1 -= 4;
-
-                        query.response[3] = 0xFF;
-                        query.response[4] = b1;
-                        query.response[5] = sum->sticks[1].horiz / 2;
-                        query.response[6] = sum->sticks[1].vert / 2;
-                        query.numBytes = 7;
-                        query.lastByte = 1;
-                        DEBUG_OUT(pad->mode);
-                        return pad->mode;
+        if (pad->config || value == 0x42 || value == 0x43) {
```

Comment:
> don't know whether you need to have query.numBytes = 0; and query.queryDone = 1; too.

Those are necessary.

I used the early return in another build. Not sure why I didn't change it before pushing this PR.